mpitests/test_apply.py

```diff
@@ -0,0 +1,480 @@
+
+    #BasisState test
+    @pytest.mark.parametrize("operation", [qml.BasisState])
+    @pytest.mark.parametrize("index", range(numQubits))
+    def test_state_prep(self, tol, operation, index):
+        par = np.zeros(numQubits,dtype=int)
+        par[index]=1
+        num_wires = numQubits
+        comm = MPI.COMM_WORLD
+        commSize = comm.Get_size()
+        num_global_wires = commSize.bit_length() - 1
+        num_local_wires = num_wires - num_global_wires
+
+        state_vector = np.zeros(1 << num_wires).astype(np.complex128)
+        expected_output_cpu = np.zeros(1 << num_wires).astype(np.complex128)
+        local_state_vector = np.zeros(1 << num_local_wires).astype(np.complex128)
+        local_expected_output_cpu = np.zeros(1 << num_local_wires).astype(np.complex128)
+
+        state_vector = createRandomInitState(num_wires)
+
+        comm.Scatter(state_vector, local_state_vector, root=0)
+        dev_cpu = qml.device("default.qubit", wires=num_wires, c_dtype=np.complex128)
+
+        @qml.qnode(dev_cpu)
+        def circuit():
+            operation(par, wires=range(numQubits))
+            return qml.state()
+
+        expected_output_cpu = circuit()
+        comm.Scatter(expected_output_cpu, local_expected_output_cpu, root=0)
+
+        dev_gpumpi = qml.device(
+            "lightning.gpu", wires=num_wires, mpi=True, c_dtype=np.complex128
+        )
+
+        @qml.qnode(dev_gpumpi)
+        def circuit_mpi():
+            operation(par, wires=range(numQubits))
+            return qml.state()
+
+        local_state_vector = circuit_mpi()
+
+        assert np.allclose(local_state_vector, local_expected_output_cpu, atol=tol, rtol=0)
+
+    test_qubit_state_prep = [
+        (qml.QubitStateVector, np.array([1 / np.sqrt(2), 1 / np.sqrt(2)]), [0]),
+        (qml.QubitStateVector, np.array([1 / np.sqrt(2), 1 / np.sqrt(2)]), [1]),
+        (qml.QubitStateVector, np.array([1 / np.sqrt(2), 1 / np.sqrt(2)]), [2]),
+        (qml.QubitStateVector, np.array([1 / np.sqrt(2), 1 / np.sqrt(2)]), [3]),
+        (qml.QubitStateVector, np.array([1 / np.sqrt(2), 1 / np.sqrt(2)]), [4]),
+        (qml.QubitStateVector, np.array([1 / np.sqrt(2), 1 / np.sqrt(2)]), [5]),
+        (qml.QubitStateVector, np.array([0, 1 / np.sqrt(2), 0, 1 / np.sqrt(2)]), [1, 0]),
+        (qml.QubitStateVector, np.array([0, 1 / np.sqrt(2), 0, 1 / np.sqrt(2)]), [0, 1]),
+        (qml.QubitStateVector, np.array([0, 1 / np.sqrt(2), 0, 1 / np.sqrt(2)]), [0, 2]),
+        (
+            qml.QubitStateVector,
+            np.array([0, 1 / np.sqrt(2), 0, 1 / np.sqrt(2)]),
+            [numQubits - 2, numQubits - 1],
+        ),
+        (
+            qml.QubitStateVector,
+            np.array([0, 1 / np.sqrt(2), 0, 1 / np.sqrt(2)]),
+            [0, numQubits - 1],
+        ),
+        (
+            qml.QubitStateVector,
+            np.array([0, 1 / np.sqrt(2), 0, 1 / np.sqrt(2)]),
+            [0, numQubits - 2],
+        ),
+    ]
```

Comment:
Since the operation is always the same, you can either add it as a pytest parameter (double parametrization technique) or add it internally to the test.